website/docs/extensions/index.md

```diff
@@ -1,21 +1,54 @@
 ---
 sidebar_position: 110
 title: Extensions
-description: Installing, developing or publishing extensions.
+description: Installing, developing, or publishing extensions.
 keywords: [podman desktop, podman, extension]
 tags: [migrating-to-kubernetes]
 ---
 
 # Podman Desktop extensions
 
-Podman Desktop extensions contribute to:
+Podman Desktop extensions are designed to be modular and easily pluggable into Podman Desktop. They allow you to use and manipulate a wide range of Podman Desktop functionality [via our API](/api). Not only can you customize every part of Podman Desktop, but you can also extend its functionality further with extensions.
 
-- Container engine support, such as Podman, Docker.
-- Virtual machine support, such as Lima.
-- Podman Desktop extension points such as tray icon menu, status bar items, icons, menus, and commands.
-- Integration with third parties tools, such as Kind or Compose.
+With extensions, you can add:
 
-#### Next steps
+- Support for container engines such as Podman or Docker.
+- Virtual machine integrations, such as Lima.
+- Podman Desktop extension points, including tray icon menus, status bar items, icons, menus, and commands.
+- Integration with third-party tools, such as Kind or Compose.
+
+## What can a Podman Desktop extension do?
+
+Here are some examples of what you can achieve with the Podman Desktop extension API:
+
+- [Create your own onboarding workflow.](/docs/extensions/developing/onboarding-workflow)
+- [Add non-native Podman Desktop commands.](/docs/extensions/developing/commands)
+- [Create configuration settings for your extension.](/docs/extensions/developing/config)
+- [Add custom commands to areas such as pushing images.](/docs/extensions/developing/menu)
+
+The possibilities are endless. You can leverage [our API](/api) to expand your extension's capabilities even further.
+
+## How to build an extension
+
+To help you get started, we've provided templates ranging from a minimal "hello world" example to a full web-view extension.
+
+Here are some examples from [our templates documentation](/docs/extensions/templates):
+
+- [Basic "Hello World" example.](https://github.com/podman-desktop/extension-template-minimal) 
+- [Simple webview template.](https://github.com/podman-desktop/extension-template-webview)
+- [Full-stack webview template.](https://github.com/podman-desktop/extension-template-full)
+
+Have questions or need assistance? Join our community on Discord for support!
+
+## Architecture
```

Comment:
We can place this Architecture section in the beginning to help readers understand the architecture before going into usage details.

- Architecture 
- What can a PD extension do?
- How to build an extension
- Next steps